src/components/app/xswap/XswapModal.tsx

```diff
@@ -0,0 +1,109 @@
+import { useAccount, useWalletClient } from 'wagmi';
+import { openTransactionModal, renderTxStatus, Transactions } from "@xswap-link/sdk";
+import { useState, useRef, useEffect } from 'react';
+import { waitForTransactionReceipt } from 'viem/actions';
+
+export const XPayButton = () => {
+  const { address, chain, connector } = useAccount();
+  const { data: walletClient } = useWalletClient();
+  const [isHovered, setIsHovered] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const timeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+  useEffect(() => {
+    return () => {
+      if (timeoutRef.current) {
+        clearTimeout(timeoutRef.current);
+      }
+    };
+  }, []);
+
+  const handleOpenXPay = async () => {
+    setIsLoading(true);
+    
+    timeoutRef.current = setTimeout(() => {
+      setIsLoading(false);
+    }, 5000);
+
+    const integratorId = process.env.NEXT_PUBLIC_INTEGRATOR_ID;
```

Comment:
If process.env.NEXT_PUBLIC_INTEGRATOR_ID should always be defined, I would rather explain it on env.mjs.
We don't need to check if NEXT_PUBLIC_INTEGRATOR_ID exists on .env then.

Reply:
good suggestion. I'm changing it.